src/prefect/filesystems.py

```diff
@@ -791,47 +793,103 @@ class GitHub(ReadableDeploymentStorage):
         default=None,
         description="An optional reference to pin to; can be a branch name or tag.",
     )
+    access_token: Optional[SecretStr] = Field(
```

Comment:
Should we try loading the `GITHUB_TOKEN` environment variable if not set?

Reply:
Is this a common pattern? I see a `GITHUB_TOKEN` for our release notes. My gut is against it because we aren't offering environment variable credentials elsewhere for storage.

Reply:
Because a block ultimately persists its state in the DB, we should avoid implicit assignment of properties.